sky/job/controller.py

```diff
@@ -292,22 +294,23 @@ def _run_one_task(self, task_id: int, task: 'sky.Task') -> bool:
             if handle is not None:
                 resources = handle.launched_resources
                 assert resources is not None, handle
-                if resources.need_cleanup_after_preemption():
+                if (resources.use_spot and
+                        resources.need_cleanup_after_preemption()):
```

Comment:
Should we do this for ondemand as well? iiuc it also needs cleanup after an unexpected error

Reply:
Kinda have the same feeling. If future resources.need_cleanup_after_preemption() impl added an on-demand case, this code won't automatically handle it?